modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java

```diff
@@ -0,0 +1,506 @@
+                            pendingDataOperations);
+                }
+            }
+        }
+    }
+
+    /**
+     * Get pending operations.
+     *
+     * @param syncmlDocument SyncmlDocument object which creates from the syncml engine using syncml payload
+     * @return Return list of pending operations.
+     * @throws OperationManagementException
+     * @throws DeviceManagementException
+     * @throws FeatureManagementException
+     * @throws PolicyComplianceException
+     * @throws NotificationManagementException
+     */
+    public List<? extends Operation> getPendingOperations(SyncmlDocument syncmlDocument)
+            throws OperationManagementException, DeviceManagementException, FeatureManagementException,
+            PolicyComplianceException, NotificationManagementException, WindowsDeviceEnrolmentException,
+            WindowsOperationException {
+
+
+        List<? extends Operation> pendingOperations;
+        DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(
+                syncmlDocument.getHeader().getSource().getLocURI());
+        UpdateUriOperations(syncmlDocument);
+        generateComplianceFeatureStatus(syncmlDocument);
+
+        pendingOperations = SyncmlUtils.getDeviceManagementService().getPendingOperations(deviceIdentifier);
+        return pendingOperations;
+    }
+
+    /**
+     * Set compliance of the feature according to the device status for the specific feature.
+     *
+     * @param activeFeature
+     * @param deviceFeature
+     * @return Returns setting up compliance feature.
+     */
+    public ComplianceFeature setComplianceFeatures(ProfileFeature activeFeature, Profile deviceFeature) {
+        ComplianceFeature complianceFeature = new ComplianceFeature();
+        complianceFeature.setFeature(activeFeature);
+        complianceFeature.setFeatureCode(activeFeature.getFeatureCode());
+        complianceFeature.setCompliance(deviceFeature.isCompliance());
+        return complianceFeature;
+    }
+
+    /**
+     * Update the completed/Error status of the operation which have the URI of the operation code in the syncml payload.
+     *
+     * @param syncmlDocument SyncmlDocument object generated from the the syncml engine.
+     * @throws DeviceManagementException
+     * @throws NotificationManagementException
+     * @throws OperationManagementException
+     */
+    public void UpdateUriOperations(SyncmlDocument syncmlDocument) throws DeviceManagementException,
+            NotificationManagementException, OperationManagementException, WindowsOperationException {
+        DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(
+                syncmlDocument.getHeader().getSource().getLocURI());
+        List<Status> statuses = syncmlDocument.getBody().getStatus();
+        OperationUtils operationUtils = new OperationUtils();
+
+        for (Status status : statuses) {
+
+            if (status.getCommand().equals(Constants.EXECUTE)) {
```

Comment:
As a best practice, always compare the non-deterministic variable values against known constants.